content/docs/command-reference/remote/modify.md

```diff
@@ -310,19 +310,89 @@ $ dvc remote modify myremote endpointurl \
   ```
 
-- `connection_string` - connection string:
+> The options below contain sensitive user info. Therefore, it's safer to use
+> the `--local` option so they're written to a Git-ignored
+> [config-file](https://dvc.org/doc/command-reference/config).
+
+- `connection_string` -
+  [connection string](http://azure.microsoft.com/en-us/documentation/articles/storage-configure-connection-string/):
 
   ```dvc
   $ dvc remote modify --local myremote connection_string \
                               'mystring'
   ```
 
-> The connection string contains sensitive user info. Therefore, it's safer to
-> add it with the `--local` option, so it's written to a Git-ignored config
-> file.
+- `account_name` - storage account name (requires either `account_key` or
+  `sas_token` along with this):
+
+  ```dvc
+  $ dvc remote modify --local myremote account_name 'myuser'
+  ```
+
+* `account_key` - storage account key (for `account_name`):
+
+  ```dvc
+  $ dvc remote modify --local myremote account_key 'mykey'
+  ```
+
+* `sas_token` - shared access signature token (can be used for `account_name`
+  instead of `account_key`):
+
+  ```dvc
+  $ dvc remote modify --local myremote sas_token 'mytoken'
+  ```
+
+* `tenant_id` - tenant ID for
+  [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal)
+  authentication:
+
+  ```dvc
+  $ dvc remote modify --local myremote tenant_id 'directory-id'
+  ```
+
+* `client_id` - client ID for _service principal_ authentication:
+
+  ```dvc
+  $ dvc remote modify --local myremote client_id 'client-id'
+  ```
+
+* `client_secret` - client Secret for _service principal_ authentication:
+
+  ```dvc
+  $ dvc remote modify --local myremote client_secret 'client-secret'
+  ```
+
+Azure remotes can also be configured entirely via environment variables:
+
+```dvc
+$ export AZURE_STORAGE_ACCOUNT_NAME='myuser'
+$ export AZURE_STORAGE_ACCOUNT_KEY='mykey'
+$ dvc remote add -d myremote azure://mycontainer/path
+```
+
+For using service principal with certificate:
+
+```dvc
+$ export AZURE_TENANT_ID='directory-id'
+$ export AZURE_CLIENT_ID='client-id'
+$ export AZURE_CLIENT_CERTIFICATE_PATH='/path/to/certificate'
+```
+
+For logging in with username/password;
+
+```
+$ export AZURE_CLIENT_ID='client-id'
+$ export AZURE_USERNAME='myuser'
+$ export AZURE_PASSWOR='mysecret'
+```
+
+> On Windows, if none of the above settings are configured the login will
+> fallback to search for a signed in Microsoft application (e.g Visual Studio)
+> and use that identity. If multiple found, it will use `AZURE_USERNAME` to
+> select the primary one.
 
-For more information on configuring Azure Storage connection strings, visit
-[here](https://docs.microsoft.com/en-us/azure/storage/common/storage-configure-connection-string).
+> On all other systems, it will use the Microsoft account of user who is signed
+> in to Visual Studio Code, if it is available.
```

Comment:
Should we put some of this default authentication info into `remote add` as well? And near the top here?

A single note (md block quote) for both paragraphs above should be OK 👍

Reply:
Or the list of order in which creds are looked for as you guys discussed in https://github.com/iterative/dvc.org/pull/2226#pullrequestreview-595535655.

Reply:
I couldn't figure out the proper place to put this, though this is what I got as a list;

```
Authentication Flow:
- If `connection_string` specified, it will auth with it
- If `tennant_id`/`client_id`/`client_secret` is specified, it will auth with AD service principal
- If `account_name`/`account_key` is specified, it will auth with account_key
- If `account_name`/`sas_token` is specified, it will auth with SAS token
- If `account_name` is specified, it will auth anonymously
- If nothing is specified in the config, it will try to follow the steps here and try to infer it from the environment (
https://azuresdkdocs.blob.core.windows.net/$web/python/azure-identity/1.4.0/azure.identity.html#azure.identity.DefaultAzureCredential)
```

Reply:
Great, thanks.

- Where in that order do env vars go? (Please mention it in the list.)
- Let's put that list before the note about "The authentication values below will contain sensitive..." please, and I'll copy-edit it.
- Please move the `tennant_id`/`client_id`/`client_secret` config options before `account_name` to match this list.

Reply:
And for the `remote add` ref lets please change:

```diff
- See `dvc remote modify` for a full list of Azure storage parameters.
+ See `dvc remote modify` for a full list of Azure storage authentication methods.
```

since all the Azure remote params are for auth.

Reply:
> Where in that order do env vars go? (Please mention it in the list.)

It is the same order.

Reply:
@isidentical I meant are env vars read first or do the config values get the priority?

Reply:
> I meant are env vars read first or do the config values get the priority?

Config values get the priority.